# Awesome Papers when Single Cell Meets Deep Learning
Must-read papers on deep learning based methodology paper for single cell data analysis.



## Contributing
Please feel free to [pull requests](https://github.com/sallyqus/SingleCellMeetsDeepLearning/pulls), email Sally (sally.qus@gmail.com) or join our chats to add links.

[![Join the chat at https://gitter.im/kjw0612/awesome-rnn](https://badges.gitter.im/Join%20Chat.svg)](https://gitter.im/kjw0612/awesome-rnn?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge)

## [Content](#content)

<table>
<tr><td colspan="2"><a href="#review-papers">1. Reviews</a></td></tr> 
<tr><td colspan="2"><a href="#dnns">2. Deep Neural Networks </a></td></tr>
<tr>
    <td>&emsp;<a href="#anns">2.1 ANNs </a></td>
    <td>&ensp;<a href="#cnns">2.2 CNNs </a></td>
</tr>
<tr>
    <td>&emsp;<a href="#others">2.3 Others </a></td>
    <td></td>
</tr>
<tr><td colspan="2"><a href="#generative-models">3. Generative Models </a></td></tr> 
<tr>
    <td>&emsp;<a href="#gan">3.1 GANs </a></td>
    <td>&ensp;<a href="#vae">3.2 VAEs </a></td>
</tr>
<tr><td colspan="2"><a href="#graph-neural-networks">4. Graph Neural Networks</a></td></tr> 
<tr><td colspan="2"><a href="#reinforcement-learning">5. Reinforcement Learning</a></td></tr> 
<tr><td colspan="2"><a href="#natural-language-processing">6. Natural Language Processing</a></td></tr> 
<tr><td colspan="2"><a href="#bayesian-deep-learning">7. Bayesian Deep Learning </a></td></tr>
</table>



## [Review papers](#content)

1. **Bayesian deep learning for single-cell analysis.** Nat. Method 2018. [paper](https://www.nature.com/articles/s41592-018-0230-9.pdf?origin=ppub)

    *Gregory P. Way, Casey S. Greene.* 
    
1. **Deep learning: new computational modelling techniques for genomics. ** Nat. Reviews Genetics 2018. [paper](https://www.nature.com/articles/s41576-019-0122-6.pdf)

    *Gökcen Eraslan, Žiga Avsec, Julien Gagneur, Fabian J. Theis*
    
1. **Emerging deep learning methods for single-cell RNA-seq data analysis.** Quantitative Biology 2019. [paper](file:///Volumes/GoogleDrive/My%20Drive/Papers/scRNAseq+Computation/2019.Zheng-Wang2019_Article_EmergingDeepLearningMethodsFor.pdf)

    *Jie Zheng, Ke Wang.*
        
## [Deep Neural Networks](#content)   

### [ANNs](#content)


### [CNNs](#content)

1. **Single-Cell Phenotype Classification Using Deep Convolutional Neural Networks.** Journal of Biomolecular Screening 2016. [paper](https://journals.sagepub.com/doi/pdf/10.1177/1087057116631284)
   
    *Oliver Dürr, Beate Sick.* 
    
## [Generative Models](#content)   


### [GAN](#content)
1. **MAGAN: Aligning Biological Manifolds.** ICML 2018. [paper](http://proceedings.mlr.press/v80/amodio18a/amodio18a.pdf)
   
    *Matthew Amodio, Smita Krishnaswamy.* 

1. **wGAN: Generative adversarial networks simulate gene expression and predict perturbations in single cells.** bioRxiv 2018. [paper](http://proceedings.mlr.press/v80/amodio18a/amodio18a.pdf)
   
    *Arsham Ghahramani, Fiona M. Watt, and Nicholas M. Luscombe.* 

1. **scGen predicts single-cell perturbation responses.** Nat. Method 2019. [paper](https://www.nature.com/articles/s41592-019-0494-8.pdf)
   
    *Mohammad Lotfollahi, F. Alexander Wolf, Fabian J. Theis.* 

1. **cscGAN: Realistic in silico generation and augmentation of single-cell RNA-seq data using generative adversarial networks.** Nat. Comm. 2020. [paper](https://www.nature.com/articles/s41467-019-14018-z.pdf)
   
    *Mohamed Marouf, Pierre Machart, Vikas Bansal, Christoph Kilian, Daniel S. Magruder, Christian F. Krebs, Stefan Bonn.* 

    

### [AE/VAE](#content)
1. **scVI: Deep generative modeling for single-cell transcriptomics.** Nat. Methods 2018. [paper](https://www.nature.com/articles/s41592-018-0229-2.pdf)
   
    *Romain Lopez, Jeffrey Regier, Michael B. Cole, Michael I. Jordan, Nir Yosef.* 
    
1. **scANVI: Probabilistic Harmonization and Annotation of Single-cell Transcriptomics Data with Deep Generative Models.** bioRxiv 2019. [paper](https://www.biorxiv.org/content/10.1101/532895v2.full.pdf)
   
    *Chenling Xu, Romain Lopez, Edouard Mehlman, Jeffrey Regier, Michael I. Jordan, Nir Yosef.* 
    
1. **SAVER-X: Data denoising with transfer learning in single-cell transcriptomics.** Nat. Methods 2019. [paper](https://www.nature.com/articles/s41592-019-0537-1.pdf)
   
    *Jingshu Wang, Divyansh Agarwal, Mo Huang, Gang Hu, Zilu Zhou, Chengzhong Ye, Nancy R. Zhang.* 
    
1. **CNNC: Deep learning for inferring gene relationships from single-cell expression data.** PNAS 2019. [paper](https://www.pnas.org/content/pnas/116/52/27151.full.pdf)
   
    *Ye Yuan, Ziv Bar-Josepha.* 
    
1. **SAUCIE: Exploring single-cell data with deep multitasking neural networks.** Nat. Methods 2019. [paper](https://www.nature.com/articles/s41592-019-0576-7.pdf)
   
    *Matthew Amodio, David van Dijk, Krishnan Srinivasan, William S. Chen, Hussein Mohsen, Kevin R. Moon, Allison Campbell, Yujiao Zhao, Xiaomei Wang, Manjunatha Venkataswamy, Anita Desai, V. Ravi, Priti Kumar, Ruth Montgomery, Guy Wolf, Smita Krishnaswamy* 
    
    

## [Graph Neural Networks](#content)  

## [Reinforcement Learning](#content)   
## [Natural Language Processing](#content)  

## [Bayesian Deep Learning](#content)   
1. **SAVER-X: Data denoising with transfer learning in single-cell transcriptomics.** Nat. Method 2019. [paper](https://www.nature.com/articles/s41592-019-0537-1.pdf)
   
    *Jingshu Wang, Divyansh Agarwal, Mo Huang, Gang Hu, Zilu Zhou, Chengzhong Ye, Nancy R. Zhang.* 

